apps/internal/base/base.go

```diff
@@ -284,6 +291,10 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	authParams.AuthorizationType = silent.AuthorizationType
 	authParams.Claims = silent.Claims
 	authParams.UserAssertion = silent.UserAssertion
+	//if validate discovery is false, add authority to known hosts, so that Alias Lookup is not done
+	if !authParams.AuthorityInfo.ValidateAuthority {
```

Comment:
I think there is a logical error here: 

- if `instance_discovery` is disabled, then `validate_authority` is also disabled (correct, since we cannot valdiate the authority)
- but not the other way around - if `validate_authority` is disabled, then it doesn't mean that the authority is known. In particular, MSALs should still perform instance discovery to find out authority aliases.

We should disable / hide / deprecate the validate_authority flag altogether, but this can be done separately. 

For now, I recommend changing this to `if !authParams.disableInstanceDiscovery == true { `